IM_calculation/test/test_im_calculation/test_im_calculation.py

```diff
@@ -287,6 +308,21 @@ def test_write_rows(self, set_up):
             sub_csv = io.StringIO()
             sub_csv_writer = csv.writer(sub_csv)
 
+            for station_n in result_dict.keys():
```

Comment:
This logic looks pretty much identical to the one above, worth making a function out of it?